src/main/java/ch/njol/skript/expressions/ExprCraftingSlots.java

```diff
@@ -0,0 +1,171 @@
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Utils;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Material;
+import org.bukkit.event.Event;
+import org.bukkit.event.inventory.PrepareItemCraftEvent;
+import org.bukkit.inventory.CraftingInventory;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Name("Crafting Inventory Slots")
+@Description({"Represents the slots/items in the crafting inventory such as the result item and matrix of the items."})
+@Examples({"on craft:",
+	"\tif crafting result item is paper:",
+	"\t\tset the crafting matrix to air, air, air, paper, diamond, paper, air, air and air",
+	"on preparing crafting:",
+	"\tset crafting result item to wood"})
+@Since("INSERT VERSION")
+public class ExprCraftingSlots extends SimpleExpression<ItemStack> {
+	
+	static {
+		Skript.registerExpression(ExprCraftingSlots.class, ItemStack.class, ExpressionType.COMBINED,
+				"[the] crafting [inventory] result (slot|item) [of %inventories%]",
+				"%inventories%'[s] crafting [inventory] result (slot|item)",
+				"[the] crafting [inventory] (matrix|grid) [(slots|items|shape)] [of %inventories%]",
+				"%inventories%'[s] crafting [inventory] (matrix|grid) [(slots|items|shape)]");
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<Inventory> invis;
+	private boolean isMatrix;
+	final ItemStack AIR = new ItemStack(Material.AIR);
+	final ItemStack[] AIR_ITEMSTACK_LIST = new ItemStack[]{AIR, AIR, AIR, AIR, AIR, AIR, AIR, AIR, AIR};
+
+	@Override
+	@SuppressWarnings({"null", "unchecked"})
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		invis = (Expression<Inventory>) exprs[0];
+		isMatrix = matchedPattern > 1;
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected ItemStack[] get(Event e) {
+		Inventory[] invis = this.invis.getArray(e);
+		if (invis == null)
+			return null;
+
+		List<ItemStack> items = new ArrayList<>(0);
+		for (Inventory invi : invis) {
+			if (!(invi instanceof CraftingInventory))
+				continue;
+
+			CraftingInventory craft = (CraftingInventory) invi;
+			if (isMatrix) {
+				items.addAll(Arrays.asList(craft.getMatrix()));
+			} else {
+				items.add(craft.getResult());
+			}
+		}
+		return items.toArray(new ItemStack[0]);
+	}
+
+	@Override
+	public @Nullable Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
```

Comment:
We don't need this switch statement here when we're only checking for 1 mode
```
if (mode == ChangeMode.SET)
    return CollectionUtils.array(ItemStack[].class);
return null;
```

Reply:
Done however, we need 2 modes SET & DELETE